### PR TITLE
SC-575-update-state-for-diffs-calculation

### PR DESCRIPTION
**Pull request by potan**
tests

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/state/diffs/InvokeScriptTransactionDiff.scala [hunk 1]
```diff
@@ -187,15 +188,15 @@ object InvokeScriptTransactionDiff {
...
-          _ <- TracedResult(
-            Either.cond(
-              transfers
-                .flatMap(_.assetId)
-                .forall(id => blockchain.assetDescription(IssuedAsset(id)).isDefined),
-              (),
-              GenericError(s"Unissued assets are not allowed")
-            )
-          )
+//          _ <- TracedResult(
```
Comment (msmolyakov) [inline node/src/main/scala/com/wavesplatform/state/diffs/InvokeScriptTransactionDiff.scala:191]: delete commented code

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala [hunk 2]
```diff
@@ -1592,6 +1593,220 @@ class InvokeScriptTransactionDiffTest
...
+  property("nonissued asset cann't be transfered") {
+    forAll(transferNonAssetIdScenario) {
+      case (invoke, genesisTxs) =>
+        tempDb { db =>
+          val features = fs.copy(
+            preActivatedFeatures = fs.preActivatedFeatures + (BlockchainFeatures.MultiPaymentInvokeScript.id -> 0)
+          )
+
+          assertDiffEi(Seq(TestBlock.create(genesisTxs)), TestBlock.create(Seq(invoke), Block.ProtoBlockVersion), features) { ei =>
+            ei shouldBe 'Left
```
Comment (msmolyakov) [inline node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1751]: What error message?

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala [hunk 3]
```diff
@@ -1592,6 +1593,220 @@ class InvokeScriptTransactionDiffTest
...
+
+  property("dublicate issuing asset should produce error") {
+    forAll(doubleAssetIdScenario) {
+      case (invoke, genesisTxs) =>
+        tempDb { db =>
+          val features = fs.copy(
+            preActivatedFeatures = fs.preActivatedFeatures + (BlockchainFeatures.MultiPaymentInvokeScript.id -> 0)
+          )
+          assertDiffEi(Seq(TestBlock.create(genesisTxs)), TestBlock.create(Seq(invoke), Block.ProtoBlockVersion), features) { ei =>
+            ei shouldBe Left(TransactionValidationError(InvalidAssetId, invoke))
```
Comment (msmolyakov) [inline node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1804]: Will it be clear to the user that an asset with this id already exists?

Comment (potan) [reply node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1804]: Which error code should we use?

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala [hunk 4]
```diff
@@ -1592,6 +1593,220 @@ class InvokeScriptTransactionDiffTest
...
+  property("Reissuind unissued asset should produce error") {
+    forAll(reissueAssetIdScenario) {
+      case (invoke, genesisTxs) =>
+        tempDb { db =>
+          val features = fs.copy(
+            preActivatedFeatures = fs.preActivatedFeatures + (BlockchainFeatures.MultiPaymentInvokeScript.id -> 0)
+          )
+
+          assertDiffEi(Seq(TestBlock.create(genesisTxs)), TestBlock.create(Seq(invoke), Block.ProtoBlockVersion), features) { ei =>
+            ei should produce("Asset is not reissuable")
```
[the remark this answers is not in the document]
Comment (potan) [reply node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1641]: Asset is exists. It's created by assetTx transaction (line 1631).

File: node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala (hunk 3 above)
Comment (msmolyakov) [reply node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1804]: maybe add a new error code like "AssetIdAlreadyExists"?

Comment (msmolyakov) [reply node/src/test/scala/com/wavesplatform/state/diffs/ci/InvokeScriptTransactionDiffTest.scala:1804]: or just describe, why this Id is invalid

**Pull request merged**